components/datetime/src/scaffold/calendar.rs

```diff
@@ -277,12 +277,14 @@ where
 
 pub(crate) struct AnyCalendarProvider<H, P> {
     provider: P,
+    // This is one of Buddhist, Chinese, Coptic, Dangi, Ethiopian, EthiopianAmeteAlem, Gregorian, Hebrew, Indian,
+    // HijriCivil, HijriObservationalMecca, HijriTabular, HijriUmmAlQura, Japanese, JapaneseExtended, Persian, Roc
     kind: AnyCalendarKind,
     _helper: PhantomData<H>,
 }
 
 impl<H, P> AnyCalendarProvider<H, P> {
-    pub(crate) fn new(provider: P, kind: AnyCalendarKind) -> Self {
+    pub(crate) fn new_unchecked(provider: P, kind: AnyCalendarKind) -> Self {
```

Comment:
Suggestion: you could make this `try_new -> Option<Self>` and do the match in here. Less coding at a distance. I don't really like having a match in the middle of `neo.rs` which is supposed to be a thin API wrapper file.

Reply:
I did this in a different way in #6352